### PR TITLE
Use for CI reusable workflows for DRY

### PR DESCRIPTION
**Pull request by spirosoik**
#### Summary
We can use across all the repositories of plugins and apps the same reusable workflow which give us the ability to change only one file instead of multiple and apart from this the ability to use different versions as we prefer.

#### Ticket Link
OSF

### CODE REVIEW

**Comment by spirosoik** [comment]
@mickmister this is even better actually. 1 re-usable CI workflow across everything which utilises the github actions ;)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: ci
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
Comment (mickmister) [inline .github/workflows/ci.yml:6]: Can we have this run on all branches? I see no harm in "over-running" the github actions

```suggestion
  push:
    branches:
      - '**'
  pull_request:
    branches:
      - '**'
```

Comment (spirosoik) [reply .github/workflows/ci.yml:6]: yeah we can but it was good to not burn time or resources for every push. I am ok if you want to enable it.

Comment (mickmister) [reply .github/workflows/ci.yml:6]: @spirosoik How does this work on GitHub actions, I mean the cost of CI runs/time/resources? These plugin repos are not as active as e.g. server and webapp, so I don't think we'll be over-using the CI resources

Comment (spirosoik) [reply .github/workflows/ci.yml]: https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions

Comment (spirosoik) [reply .github/workflows/ci.yml]: feel free to raise a PR in the actions workflows accordingly.

**Comment by spirosoik** [comment]
@phoinixgrr please have a look

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+
+permissions:
+  contents: read
+
+jobs:
+  plugin-cd:
+    uses: mattermost/actions-workflows/.github/workflows/plugin-cd.yml@11369c571c4a721bfaca5d9c7f3920c201e79a20
+    secrets: inherit
```
Comment (phoinixgrr) [inline .github/workflows/cd.yml:17]: noice 👍

**Review by phoinixgrr — APPROVED** [review]
Its beautiful. 
Great Work overall @spirosoik  🎖️ 🏅 🥇

**Pull request merged**